hare/hare.go

```diff
@@ -470,7 +470,24 @@ func goodProposals(ctx context.Context, logger log.Log, msh mesh, nodeID types.N
 	if ownHdr != nil {
 		ownTickHeight = ownHdr.TickHeight()
 	}
+	atxs := map[types.ATXID]int{}
 	for _, p := range props {
+		atxs[p.AtxID]++
+	}
+	for _, p := range props {
+		if p.IsMalicious() {
+			logger.With().Warning("not voting on proposal from malicious identity",
+				log.Stringer("id", p.ID()),
+			)
+			continue
+		}
+		if atxs[p.AtxID] > 1 {
```

Comment:
Why do we need this check?

Won't the proposal be set to malicious when being retrieved from the mesh if there were multiple ballots from the same smesher?

Reply:
it doesn't do that if smesher registered several proposals with same ballot. that should be fixed.
but i would prefer to have this check regardless of that, as this such bug is too expensive

Reply:
Ahh ok so this is the case of multiple proposals with one good ballot? So I guess we'll need a new malfeasance proof type for this?

Reply:
or simply reject proposal, but in my mind it is more important to be safe here